UKHO.ADM/internal/ConvertTo-GroupObjects.ps1

```diff
@@ -28,16 +38,27 @@ function ConvertTo-GroupObjects {
                 $ggGroup = [ADGroup]::new($Settings.GroupPrefix, $Settings.Environment, $OU.Name, $GroupConfig.Name, "GG", $OU.DistinguishedName, $OU.Domain)
                 $ugGroup.ADGroupMembers += $ggGroup
 
+                # Add Other Groups
+                if ($null -ne $GroupConfig.Groups -and $GroupConfig.Groups.Count -gt 0) {
+                    $GroupConfig.Groups | ForEach-Object {
+                        $groupGroup = [ADGroup]::new($_.DistinguishedName, $_.Domain)
+                        if (Check-GroupExists $groupGroup) {
+                            $ugGroup.ADGroupMembers += $groupGroup
+                        }
+                    }
+                }
+
                 $groups += $ugGroup
                 $groups += $ggGroup
             }
 
             #Add users
-            if ($GroupConfig.Users -ne $null -and $GroupConfig.Users.Count -gt 0) { 
-                $GroupConfig.Users | Where {$_.Domain -eq $OU.Domain.FQDN} | ForEach-Object {
+            if ($null -ne $GroupConfig.Users -and $GroupConfig.Users.Count -gt 0) { 
```

Comment:
Why the inversion for `$var -ne $null` to `$null -ne $var`?

Reply:
PS lint on VS code suggests that as a problem

Reply:
https://rencore.com/blog/powershell-null-comparison/

tl;dr
Use $null in comparison on the left hand side. Due to type inference Arrays will be checked for $null with each elements total null count equating.

Reply:
Huh. OK. Fair enough, if that is what is PSScripAnalyser suggests.